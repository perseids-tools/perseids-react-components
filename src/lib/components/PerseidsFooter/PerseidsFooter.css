html {
  position: relative;
  min-height: 100%;
  padding-bottom: 62px;
}

.perseids-react-components--footer {
  position: absolute;
  bottom: 0;
  width: 100%;
  height: 62px;
}

.perseids-react-components--footer-nav {
  height: 100%;
  flex-wrap: nowrap;
}

.perseids-react-components--doi {
  height: 30px;
}

.perseids-react-components--twitter {
  height: 30px;
}

.perseids-react-components--octicon {
  height: 30px;
}
