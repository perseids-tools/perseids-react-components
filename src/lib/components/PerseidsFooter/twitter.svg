<svg width="20" height="20" xmlns="http://www.w3.org/2000/svg">
  <rect height="24" width="24" fill="none" x="0"/>
  <g>
   <path id="svg_3" d="m20.01125,3.799c-0.736,0.326 -1.527,0.547 -2.357,0.646c0.847,-0.508 1.498,-1.312 1.804,-2.27c-0.793,0.47 -1.67,0.812 -2.606,0.996c-0.746,-0.798 -1.813,-1.296 -2.993,-1.296c-2.266,0 -4.103,1.837 -4.103,4.103c0,0.322 0.036,0.635 0.106,0.935c-3.41,-0.17 -6.433,-1.804 -8.457,-4.287c-0.353,0.607 -0.556,1.312 -0.556,2.064c0,1.424 0.724,2.68 1.825,3.415c-0.673,-0.022 -1.305,-0.207 -1.86,-0.514l0,0.052c0,1.988 1.415,3.647 3.293,4.023c-0.344,0.095 -0.707,0.145 -1.08,0.145c-0.265,0 -0.522,-0.026 -0.773,-0.074c0.522,1.63 2.038,2.817 3.833,2.85c-1.404,1.1 -3.174,1.757 -5.096,1.757c-0.332,0 -0.66,-0.02 -0.98,-0.057c1.816,1.164 3.973,1.843 6.29,1.843c7.547,0 11.675,-6.252 11.675,-11.675c0,-0.178 -0.004,-0.355 -0.012,-0.53c0.802,-0.578 1.497,-1.3 2.047,-2.124l0,-0.002z"/>
 </g>
</svg>
