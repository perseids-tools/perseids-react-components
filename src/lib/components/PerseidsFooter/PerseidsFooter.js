import React from 'react';
import PropTypes from 'prop-types';

import './PerseidsFooter.css';

import githubIcon from 'octicons/build/svg/mark-github.svg';
import reportIcon from 'octicons/build/svg/report.svg';
import twitterIcon from './twitter.svg';

const renderCopyright = text => (
  <span className="navbar-text">
    {text}
  </span>
);

const renderOcticon = (href, img, title, alt) => (
  <li className="nav-item">
    <a className="nav-link p-2" href={href} target="_blank" rel="noopener noreferrer">
      <img className="perseids-react-components--octicon" src={img} title={title} alt={alt} />
    </a>
  </li>
);

const renderTwitter = href => (
  <li className="nav-item">
    <a className="nav-link p-2" href={href} target="_blank" rel="noopener noreferrer">
      <img className="perseids-react-components--twitter" src={twitterIcon} title="Twitter" alt="Twitter Icon" />
    </a>
  </li>
);

const PerseidsFooter = ({
  children,
  copyright,
  github,
  report,
  twitter,
}) => (
  <footer className="footer perseids-react-components--footer">
    <nav className="navbar navbar-light bg-light py-0 perseids-react-components--footer-nav">
      {!!copyright && renderCopyright(copyright)}

      {children}

      <ul className="navbar-nav my-2 my-lg-02 flex-row">
        {!!report && renderOcticon(report, reportIcon, 'Report Issue', 'Report Icon')}
        {!!github && renderOcticon(github, githubIcon, 'View Source on Github', 'GitHub Icon')}
        {!!twitter && renderTwitter(twitter)}
      </ul>
    </nav>
  </footer>
);

PerseidsFooter.defaultProps = {
  children: undefined,
  copyright: 'Â© The Perseids Project 2018',
  github: 'https://github.com/perseids-project',
  report: '',
  twitter: 'https://twitter.com/PerseidsProject',
};

PerseidsFooter.propTypes = {
  children: PropTypes.node,
  copyright: PropTypes.string,
  github: PropTypes.string,
  report: PropTypes.string,
  twitter: PropTypes.string,
};

export default PerseidsFooter;
