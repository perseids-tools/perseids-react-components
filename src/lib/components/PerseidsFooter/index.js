import PerseidsFooter from './PerseidsFooter';

export default PerseidsFooter;
