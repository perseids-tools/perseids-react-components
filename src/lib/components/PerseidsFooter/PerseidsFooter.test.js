import React from 'react';
import renderer from 'react-test-renderer';

import PerseidsFooter from './PerseidsFooter';

it('renders with copyright text and default GitHub and Twitter links', () => {
  const tree = renderer.create(<PerseidsFooter />).toJSON();

  expect(tree).toMatchSnapshot();
});

it('renders custom copyright notice and  GitHub and Twitter links', () => {
  const component = (
    <PerseidsFooter
      copyright="Â© Example"
      github="https://example.com/github"
      twitter="https://example.com/twitter"
    />
  );
  const tree = renderer.create(component).toJSON();

  expect(tree).toMatchSnapshot();
});

it('can not render copyright notice and GitHub and Twitter links', () => {
  const component = (
    <PerseidsFooter copyright="" github="" twitter="" />
  );
  const tree = renderer.create(component).toJSON();

  expect(tree).toMatchSnapshot();
});

it('can render report link', () => {
  const component = (
    <PerseidsFooter github="" twitter="" report="https://example.com/report" />
  );
  const tree = renderer.create(component).toJSON();

  expect(tree).toMatchSnapshot();
});

it('can render DOI link', () => {
  const component = (
    <PerseidsFooter github="" twitter="" doi="https://example.com/doi" />
  );
  const tree = renderer.create(component).toJSON();

  expect(tree).toMatchSnapshot();
});

it('can render children between copyright notice and icons', () => {
  const component = (
    <PerseidsFooter>
      Footer text
    </PerseidsFooter>
  );
  const tree = renderer.create(component).toJSON();

  expect(tree).toMatchSnapshot();
});
