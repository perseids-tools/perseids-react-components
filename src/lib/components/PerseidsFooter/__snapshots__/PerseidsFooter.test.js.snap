// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`can not render copyright notice and GitHub and Twitter links 1`] = `
<footer
  className="footer perseids-react-components--footer"
>
  <nav
    className="navbar navbar-light bg-light py-0 perseids-react-components--footer-nav"
  >
    <ul
      className="navbar-nav my-2 my-lg-02 flex-row"
    />
  </nav>
</footer>
`;

exports[`can render children between copyright notice and icons 1`] = `
<footer
  className="footer perseids-react-components--footer"
>
  <nav
    className="navbar navbar-light bg-light py-0 perseids-react-components--footer-nav"
  >
    <span
      className="navbar-text"
    >
      © The Perseids Project 2018
    </span>
    Footer text
    <ul
      className="navbar-nav my-2 my-lg-02 flex-row"
    >
      <li
        className="nav-item"
      >
        <a
          className="nav-link p-2"
          href="https://github.com/perseids-project"
          rel="noopener noreferrer"
          target="_blank"
        >
          <img
            alt="GitHub Icon"
            className="perseids-react-components--octicon"
            src="mark-github.svg"
            title="View Source on Github"
          />
        </a>
      </li>
      <li
        className="nav-item"
      >
        <a
          className="nav-link p-2"
          href="https://twitter.com/PerseidsProject"
          rel="noopener noreferrer"
          target="_blank"
        >
          <img
            alt="Twitter Icon"
            className="perseids-react-components--twitter"
            src="twitter.svg"
            title="Twitter"
          />
        </a>
      </li>
    </ul>
  </nav>
</footer>
`;

exports[`can render report link 1`] = `
<footer
  className="footer perseids-react-components--footer"
>
  <nav
    className="navbar navbar-light bg-light py-0 perseids-react-components--footer-nav"
  >
    <span
      className="navbar-text"
    >
      © The Perseids Project 2018
    </span>
    <ul
      className="navbar-nav my-2 my-lg-02 flex-row"
    >
      <li
        className="nav-item"
      >
        <a
          className="nav-link p-2"
          href="https://example.com/github"
          rel="noopener noreferrer"
          target="_blank"
        >
          <img
            alt="Report Icon"
            className="perseids-react-components--octicon"
            src="report.svg"
            title="Report Issue"
          />
        </a>
      </li>
    </ul>
  </nav>
</footer>
`;

exports[`renders custom copyright notice and  GitHub and Twitter links 1`] = `
<footer
  className="footer perseids-react-components--footer"
>
  <nav
    className="navbar navbar-light bg-light py-0 perseids-react-components--footer-nav"
  >
    <span
      className="navbar-text"
    >
      © Example
    </span>
    <ul
      className="navbar-nav my-2 my-lg-02 flex-row"
    >
      <li
        className="nav-item"
      >
        <a
          className="nav-link p-2"
          href="https://example.com/github"
          rel="noopener noreferrer"
          target="_blank"
        >
          <img
            alt="GitHub Icon"
            className="perseids-react-components--octicon"
            src="mark-github.svg"
            title="View Source on Github"
          />
        </a>
      </li>
      <li
        className="nav-item"
      >
        <a
          className="nav-link p-2"
          href="https://example.com/twitter"
          rel="noopener noreferrer"
          target="_blank"
        >
          <img
            alt="Twitter Icon"
            className="perseids-react-components--twitter"
            src="twitter.svg"
            title="Twitter"
          />
        </a>
      </li>
    </ul>
  </nav>
</footer>
`;

exports[`renders with copyright text and default GitHub and Twitter links 1`] = `
<footer
  className="footer perseids-react-components--footer"
>
  <nav
    className="navbar navbar-light bg-light py-0 perseids-react-components--footer-nav"
  >
    <span
      className="navbar-text"
    >
      © The Perseids Project 2018
    </span>
    <ul
      className="navbar-nav my-2 my-lg-02 flex-row"
    >
      <li
        className="nav-item"
      >
        <a
          className="nav-link p-2"
          href="https://github.com/perseids-project"
          rel="noopener noreferrer"
          target="_blank"
        >
          <img
            alt="GitHub Icon"
            className="perseids-react-components--octicon"
            src="mark-github.svg"
            title="View Source on Github"
          />
        </a>
      </li>
      <li
        className="nav-item"
      >
        <a
          className="nav-link p-2"
          href="https://twitter.com/PerseidsProject"
          rel="noopener noreferrer"
          target="_blank"
        >
          <img
            alt="Twitter Icon"
            className="perseids-react-components--twitter"
            src="twitter.svg"
            title="Twitter"
          />
        </a>
      </li>
    </ul>
  </nav>
</footer>
`;
